Error building on Windows

(Copied from @laudney's comment on #17)

> I've tried to follow your instructions. All seem to work flawlessly (for about 45min) until it failed with:
> 
> ```
> The target "BeforeGenerateProjectPriFile" listed in a BeforeTargets attribute at "C:\Program Files (x86)\MSBuild\Microsoft\NuGet\Microsoft.NuGet.targets (186,61)" does not exist in the project, and will be ignored.
> Done Building Project "H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default targets) -- FAILED.
> 
> "H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (1) ->
> "H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj" (default target) (3) ->
> "H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\tf_core_cpu.vcxproj" (default target) (4) ->
> "H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\tf_core_framework.vcxproj" (default target) (5) ->
> (CustomBuild target) ->
>   C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(171,5): error MSB6006: "cmd.exe" exited with code 1. [H:\PycharmProjects\t
> ensorflow\tensorflow\contrib\cmake\build\tf_core_framework.vcxproj]
> 
>     30 Warning(s)
>     1 Error(s)
> ```

@mrry New error below after 1h of compilation. Please let me know if you need the log.

"C:\src\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (1) ->
"H:\PycharmProjects\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj" (default target) (3) ->
(ClCompile target) ->
  H:\PycharmProjects\tensorflow\tensorflow\python\lib\core\py_func.cc(165): error C2466: cannot allocate an array of constant size 0 [H:\PycharmProjects\tensor
flow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]

--- a/tensorflow/python/lib/core/py_func.cc
+++ b/tensorflow/python/lib/core/py_func.cc
@@ -15,6 +15,8 @@ limitations under the License.
 
 #include "tensorflow/python/lib/core/py_func.h"
 
+#include <array>
+
 #include <Python.h>
 #include "numpy/arrayobject.h"
 #include "tensorflow/core/framework/op_kernel.h"
@@ -162,8 +164,8 @@ bool IsSingleNone(PyObject* obj) {
   if (PyArray_NDIM(array_obj) != 0 || PyArray_SIZE(array_obj) != 1) {
     return false;
   }
-  npy_intp indices[] = {};
-  char* item_ptr = static_cast<char*>(PyArray_GetPtr(array_obj, indices));
+  std::array<npy_intp, 0> indices;
+  char* item_ptr = static_cast<char*>(PyArray_GetPtr(array_obj, indices.data()));
   PyObject* item = PyArray_GETITEM(array_obj, item_ptr);
   CHECK(item);
   return item == Py_None;

